src/AbstractFixer.php

```diff
@@ -89,21 +89,13 @@ public function supports(\SplFileInfo $file)
         return true;
     }
 
-    /**
-     * {@inheritdoc}
-     */
     public function configure(array $configuration = null)
     {
         if (!$this instanceof ConfigurationDefinitionFixerInterface) {
-            throw new \LogicException('Cannot run method for class not implementing `ConfigurationDefinitionFixerInterface`.');
+            throw new \LogicException(sprintf('Cannot configure using Abstact parent, child not implementing "%s".', ConfigurationDefinitionFixerInterface::class));
         }
 
         if (null === $configuration) {
-            @trigger_error(
```

Comment:
I think this is what @keradus meant in this comment https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/2500#discussion_r100700016
Deprecating allowing `null` as default configuration is not needed in this PR. If we want to deprecate it we can do this through a separate PR and see what people think of it there. Removing it here will leave less to discuss, meaning we can go forward faster ;)